File: AngularProject1/src/app/models/section.ts
import { TeacherDto } from "./teacherDto";
import { GetCourseDto } from "./course";

export class SectionForAddUpdate {
  id: number;
  name: string;
  courseId: number;
  teacherId: number;
  totalHours: number;
  lecturesPerWeek: number;
  timeFrom: number;
  timeto:number;


}


export class SectionData {
  id: number;
  name: string;
  totalHours: number;
  lecturesPerWeek: number;
  timeFrom: number;
  timeto: number;
  teacher: TeacherDto;
  course: GetCourseDto;

}

File: API/Models/Domain/Section.cs
namespace API.Models.Domain
{
    public class Section
    {
        public int Id { get; set; }
            
        public string Name { get; set; } = string.Empty;

        public virtual Course? Course { get; set; } 

        public virtual int CourseId { get; set; }

        public virtual User? Teacher { get; set; } 

        public virtual int TeacherId { get; set; }

        public int? TimeFrom { get; set; }

        public int? TimeTo { get; set; }

        public int TotalHours { get; set; }

        public int LecturesPerWeek { get; set; }





        //number of hours 80h
    }
}
